src/Compiler/Driver/ReuseTcResults/ReuseTcResultsDriver.fs

```diff
@@ -0,0 +1,124 @@
+module internal FSharp.Compiler.ReuseTcResults
+
+open System
+open System.Collections.Generic
+open System.IO
+
+open FSharp.Compiler.CompilerConfig
+open FSharp.Compiler.Diagnostics
+open FSharp.Compiler.GraphChecking
+open FSharp.Compiler.IO
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Syntax.PrettyNaming
+
+open Internal.Utilities.Hashing
+
+let (++) a b = Path.Combine(a, b)
+
+[<Sealed>]
+type ReuseTcResultsDriver(tcConfig: TcConfig) =
+
+    let outputDir = tcConfig.outputDir |> Option.defaultValue ""
+    let tcDataFolderName = outputDir ++ FSharpTcDataResourceName
+
+    let cmdLineFilePath = tcDataFolderName ++ "cmdline"
+    let graphFilePath = tcDataFolderName ++ "graph"
+    let referencesFilePath = tcDataFolderName ++ "references"
```

Comment:
I don't think those should be separate files, to avoid partial and incorrect data, it should be as atomic as possible.

Reply:
That was my first idea, but than I thought - granularity adds some flexibility here: if one of those things changes, we won't need to rewrite all of them, saving some time and memory. Multiple files are also a bit easier to work with - there is no need to set up some file structure here (what follows what, empty lines and so on).

What do you think?

Reply:
It risks mixing things, I would put as we do with sigdata or assembly - all in one resource. It's fine rewriting all of them, it's not a lot of data, but it is easier to verify that **all data** is valid.

Reply:
Alright then, now there's only one file.